src/server/game/Server/WorldSession.h

```diff
@@ -348,8 +348,40 @@ class WorldSession
     void SendPacket(WorldPacket const* packet);
     void SendPetNameInvalid(uint32 error, std::string const& name, DeclinedName* declinedName);
     void SendPartyResult(PartyOperation operation, std::string const& member, PartyResult res, uint32 val = 0);
-    void SendAreaTriggerMessage(const char* Text, ...) ATTR_PRINTF(2, 3);
-    void SendAreaTriggerMessage(uint32 entry, ...);
+
+    template<typename... Args>
+    void SendAreaTriggerMessage(char const* Text, Args&&... args)
+    {
+        // Use Acore::StringFormat to generate the formatted string
+        std::string formattedStr = Acore::StringFormat(Text, std::forward<Args>(args)...);
+        uint32 length = formattedStr.size() + 1;
+
+        // Create the packet and send it
+        WorldPacket data(SMSG_AREA_TRIGGER_MESSAGE, 4 + length);
+        data << length;
+        data << formattedStr;
+        SendPacket(&data);
+    }
+
+    template<typename... Args>
+    void SendAreaTriggerMessage(uint32 entry, Args&&... args)
+    {
+        // Retrieve the format string using the entry
+        char const* format = GetAcoreString(entry);
+        if (format)
+        {
+            // Use Acore::StringFormat to generate the formatted string
+            std::string formattedStr = Acore::StringFormat(format, std::forward<Args>(args)...);
+            uint32 length = formattedStr.size() + 1;
+
+            // Create the packet and send it
+            WorldPacket data(SMSG_AREA_TRIGGER_MESSAGE, 4 + length);
+            data << length;
+            data << formattedStr;
+            SendPacket(&data);
+        }
+    }
```

Comment:
Actually, I would like as little code in headerfiles as possible. So I would do something like this in the headerfile
```cpp
void SendAreaTriggerMessage(std::string_view str);

template<typename... Args>
void SendAreaTriggerMessage(char const* fmt, Args&&... args)
{
    if (HasSession())
        SendAreaTriggerMessage(Acore::StringFormat(fmt, std::forward<Args>(args)...));
}
template<typename... Args>
void SendAreaTriggerMessage(uint32 strId, Args&&... args)
{
    if (HasSession())
        SendAreaTriggerMessage(Acore::StringFormat(GetAcoreString(strId), std::forward<Args>(args)...));
}
```
and in the the cpp file I would move what you got now. so
```cpp
void WorldSession::SendAreaTriggerMessage(std::string_view str)
{
    std::vector<std::string_view> lines = Acore::Tokenize(str, '\n', true);
    for (std::string_view line : lines)
    {
        uint32 length = line.size() + 1;
        WorldPacket data(SMSG_AREA_TRIGGER_MESSAGE, 4 + length);
        data << length;
        data << line.data();
        SendPacket(&data);
    }
}

Reply:
Thanks. I was also a little bit at odds with the implementation in the h file.